apkid/rules/elf/packers.yara

```diff
@@ -372,6 +372,7 @@ rule promon_a : packer
   condition:
     is_elf and
     $rnd_libname and
+    not promon_b and
     for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncu/) and
```

Comment:
Since this is the only other rule, and the other rule is just a lib with a random looking name, how much do you know about these section names? Are they truly non-standard? Are they just something odd you found?

Also, do you think it's worth it to modify the `$s1`, `$s2`, `$s3` strings from `promon_a` to this method? This way looks better -- it's more accurate and may even be more performant since you don't need to search the entire file for a string.

Reply:
Yeah @CalebFenton. I do think that we should also modify the other lib to only match Promon-only segment names. My question was how we could match 2 out of 3 conditions using sections.name.

Reply:
I don't think there's an elegant way. I think you have to perform `C(2, 3)`:

```yara
// Match at least two section names from .ncu, .ncc, .ncd
(for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncu/) and for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncc/)) or
(for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncu/) and for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncd/)) or
(for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncc/) and for any i in (0..elf.number_of_sections): (elf.sections[i].name matches /\.ncd/))
```

Reply:
Looks ugly and slow. Lets make it with OR and only match 1 out of 3 which is good enough.

Reply:
I'm fine with how this looks and I don't think it's that much slower since `and` short-circuits. I.e. it'll only perform a single additional operation than the alternative IIF the file is promon packed.

Checking more than one section name seems very useful here since they're so short.

Reply:
Feeback taken. Thanks @CalebFenton